client/app/ui/header/header-nav.tsx

```diff
@@ -1,7 +1,84 @@
-export function HeaderNav(){
-    return(
-        <nav>
-            <p>I'm going to be the nav bar</p>
-        </nav>
+'use client'
+import { useState } from "react"
+import { Search } from "../search"
+
+export function HeaderNav() {
+
+    const [knifeDropped, setKnifeDropped] = useState(false)
+    const [creationsDropped, setCreationsDropped] = useState(false)
+
+    function toggleKnifeMenu() {
+        if (knifeDropped) {
+            setKnifeDropped(false)
+        } else {
+            setKnifeDropped(true)
+        }
+    }
```

Comment:
A trick to doing this is to `setKnifeDropped` to `!knifeDropped` like this. This isn't me telling you how to do it, btw. Suggestions are just nicely formatted, so it's a good way to communicate tips like this. If you prefer it the way you have it, that's perfectly fine  
```suggestion
    function toggleKnifeMenu() {
            setKnifeDropped(!knifeDropped)
    }
```